src/openms_ml/source/ML/AlphapeptdeepWrapper.cpp

```diff
@@ -0,0 +1,187 @@
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Timo Sachsenberg $
+//  --------------------------------------------------------------------------
+
+#include <OpenMS/ML/AlphapeptdeepWrapper.h>
+#include <OpenMS/ML/AlphaDatahandling.h>
+#include <OpenMS/ML/OpenMS_MLConfig.h>
+#include <torch/torch.h>
+#include <torch/script.h>
+#include <vector>
+using namespace OpenMS;
+
+struct Alphapeptdeepwrapper::Impl 
+{
+    torch::jit::script::Module model;
+    config_param config_data;
+  
+    Impl(const std::string& model_path, const std::string& model_config_path)
+    {
+       try 
+       {
+        	model = torch::jit::load(model_path);
+        }
+        catch (const c10::Error& e) 
+        {
+          std::cerr << "Error loading the model..\n";
+        }
+        
+        config_data = Alphapeptdeepwrapper::set_config_param(model_config_path);
+        
+    };
+    
+    ~Impl()=default;
+  
+};
+
+ 
+Alphapeptdeepwrapper::Alphapeptdeepwrapper(const std::string& model_path, const std::string& model_config_path)
+    : pimpl(new Impl(model_path, model_config_path)){}
+ 
+Alphapeptdeepwrapper::~Alphapeptdeepwrapper() = default;
+ 
+config_param Alphapeptdeepwrapper::set_config_param(const std::string& model_config_path)
+{
+    config_param data;
+    std::ifstream infile(model_config_path);
+    std::string line;
+    
+    std::string check_key = "mod_elements:";
+    while (getline(infile, line)) {
+        if (line.empty() || line[0] == '#') continue;
+
+        std::stringstream ss(line);
+        std::string key, value;
+        ss >> key >> value;
+        
+        if (check_key == "mod_elements:")
+        {
+            if (key == "-")
+            { 
+               data.mod_elements.push_back(value);
+            }
+            else
+            {
+              check_key = key;
+            }
+        } 
+        else if (check_key == "instruments:") 
+        {
+            if (key == "-")
+            { 
+               data.instruments.push_back(value);
+            }
+            else
+            {
+               check_key = key;
+            }
+            
+        } 
+        
+        if (key == "max_instrument_num:") 
+        {
+            data.max_instrument_num = std::stoi(value);
+        } 
+        else if (key == "aa_embedding_size:") 
+        {  
+            data.aa_embedding_size = std::stoi(value);
+        }
+       
+    }
+    
+    return data;
+}
+
+
+std::vector<float> Alphapeptdeepwrapper::predict(const std::vector<std::string>& seq)
+{  
+    AlphaDatahandler Data_handler(seq);
```

Comment:
The idea is, here we will pass the modified sequence as EMM(Oxidation@M)SQVTLQHMN,
so AlphaDatahandler class will care of features needed.